crates/types/src/config/object_store.rs

```diff
@@ -18,35 +18,41 @@ use serde_with::serde_as;
 #[serde(rename_all = "kebab-case")]
 #[builder(default)]
 pub struct ObjectStoreOptions {
-    /// The AWS configuration profile to use for S3 object store destinations.
+    /// The AWS configuration profile to use for S3 object store destinations. If you use
+    /// named profiles in your AWS configuration, you can replace all the other settings with
+    /// a single profile reference. See the [AWS documentation on profiles]
+    /// (https://docs.aws.amazon.com/sdkref/latest/guide/file-format.html) for more.
     #[serde(skip_serializing_if = "Option::is_none")]
     pub aws_profile: Option<String>,
 
-    /// AWS region to use with S3 object store destinations.
+    /// AWS region to use with S3 object store destinations. This may be inferred from the
+    /// environment, for example the current region when running in EC2. Because of the
+    /// request signing algorightm this must have a value. For Minio, you can generally
+    /// set this to any string, such as `us-east-1`.
     #[serde(skip_serializing_if = "Option::is_none")]
     pub aws_region: Option<String>,
 
-    /// AWS access key. We strongly recommend against using long-lived credentials; set up a
-    /// configuration profile with a role-based session credentials provider instead.
+    /// AWS access key (or username in Minio / S3-compatible stores).
     #[serde(skip_serializing_if = "Option::is_none")]
     pub aws_access_key_id: Option<String>,
 
-    /// AWS secret key. We strongly recommend against using long-lived credentials; set up a
-    /// configuration profile with a role-based session credentials provider instead.
+    /// AWS secret key (or password in Minio / S3-compatible stores).
     #[serde(skip_serializing_if = "Option::is_none")]
     pub aws_secret_access_key: Option<String>,
 
-    /// AWS session token. We strongly recommend against using long-lived credentials; set up a
-    /// configuration profile with a role-based session credentials provider instead.
+    /// AWS session token, needed with short-term STS session credentials.
```

Comment:
does this replace the need for key+secret?

Reply:
No; it's a third component in addition to key id & secret, needed when short-lived session credentials are in use.

https://docs.aws.amazon.com/IAM/latest/UserGuide/id_credentials_temp_use-resources.html

https://min.io/docs/minio/linux/developers/security-token-service.html